ArdusatSDK.cpp

```diff
@@ -453,266 +217,1101 @@ const char * valuesToCSV(const char *sensorName, unsigned long timestamp, int nu
+ * @return  sensor readings in CSV format or empty string if uninitialized
+ */
+const char * Sensor::readToCSV(const char * sensorName) {
+  this->read();
+  return this->toCSV(sensorName);
 }
 
-_DEFAULT_SENSOR_NAME_TO_CSV_FXN(luminosity)
-const char * luminosityToCSV(const char *sensorName, luminosity_t & input) {
-  _headerToCSV(&(input.header), sensorName);// warning : resets the internal buffer
+/**
+ * @brief   Takes a reading from the sensor and returns value in JSON format
+ * @ingroup sensor
+ * @param   sensorName The text to display next to the value
+ * @return  sensor readings in JSON format or empty string if uninitialized
+ */
+const char * Sensor::readToJSON(const char * sensorName) {
+  this->read();
+  return this->toJSON(sensorName);
+}
 
-  dtostrf(input.lux, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-  _add_checksum_to_csv_buffer(sensorName, 1, input.lux);
-  _getOutBuf()[_output_buf_len++] = '\n';
 
-  return _getOutBuf();
+/**************************************************************************//**
+ * @brief   Constructs Acceleration sensor object
+ * @ingroup acceleration
+ *
+ * Example Usage:
+ * @code
+ *     Acceleration accel;                        // Instantiate sensor object
+ *     accel.begin();                             // Initialize sensor
+ *     Serial.println(accel.readToJSON("accel")); // Read and print values in JSON
+ * @endcode
+ *****************************************************************************/
+Acceleration::Acceleration(void) :
+  gGain(LSM303_ACCEL_GAIN8G)
+{
+  this->initializeHeader(SENSORID_ADAFRUIT9DOFIMU, DATA_UNIT_METER_PER_SECONDSQUARED, acceleration_sensor_name);
 }
 
-_DEFAULT_SENSOR_NAME_TO_CSV_FXN(uvlight)
-const char * uvlightToCSV(const char *sensorName, uvlight_t & input) {
-  _headerToCSV(&(input.header), sensorName);// warning : resets the internal buffer
+/**
+ * @brief   Sets the gain configuration variable
+ * @ingroup acceleration
+ *
+ * @param gGain Advanced configuration for accelerometer's gain
+ *     - LSM303_ACCEL_GAIN2G
+ *     - LSM303_ACCEL_GAIN4G
+ *     - LSM303_ACCEL_GAIN6G
+ *     - LSM303_ACCEL_GAIN8G (Default)
+ *     - LSM303_ACCEL_GAIN16G
+ */
+void Acceleration::configure(lsm303_accel_gain_e gain) {
+  this->gGain = gain;
+  this->initialized = false;
+}
 
-  dtostrf(input.uvindex, 2, 3, _getOutBuf() + _output_buf_len);
-  _output_buf_len = strlen(_getOutBuf());
-  _add_checksum_to_csv_buffer(sensorName, 1, input.uvindex);
-  _getOutBuf()[_output_buf_len++] = '\n';
+/**
+ * @brief   Initializes the sensor with any set configurations or defaults
+ * @ingroup acceleration
+ *
+ * @retval true  Successfully initialized
+ * @retval false Failed to initialize
+ */
+boolean Acceleration::initialize(void) {
+  return lsm303_accel_init(this->gGain);
+}
 
-  return _getOutBuf();
+/**
+ * @brief   Takes a reading from the sensor
+ * @ingroup acceleration
+ */
+void Acceleration::read(void) {
```

Comment:
At this point, it would just be a `return initialized`. But in theory we should be able to have driver logic for whether the I2C read succeeds or errs out, which could be bubbled up in a future version.